example/test_problem/Hydro/CCSN/python_script/convert_progenitor_form.py

```diff
@@ -0,0 +1,164 @@
+    variables       = np.empty(nvars,      dtype=np.float)
+
+    outer_radius = 0.0
+    outer_vel    = 0.0
+    for line in f:
+        if ('wind:' in line):
+            break
+
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line.split()[icol])
+            # convert cell outer variables to cell centered values (radius, velocity)
+            if idx == 0:
+                variables[idx], outer_radius = 0.5 * (outer_radius + variables[idx]), variables[idx]
+            if idx == 3:
+                variables[idx], outer_vel = 0.5 * (outer_vel + variables[idx]), variables[idx]
+
+        GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM, variables])
+
+    # add a safer line for the innermost radius
+    GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM[0], GAMER_DATA_FORM])
+    GAMER_DATA_FORM[0, 0] = 1.0
+
+    return f_info, GAMER_DATA_FORM
+
+
+def convert_MESA(file):
+
+    # physical constants
+    R_sol = 6.977e10 # solar radius
+    M2CM  = 1e2      # meter to centimeter
+
+    # read the header
+    headers = []
+    f_info = ''
+    for i in range(5):
+        f.readline()
+    headers = f.readline().split()
+
+    # find and read targeting variables (radius, density, temperature, radial velocity,
+    #                                    electron fraction, pressure, and angular velocity)
+    nvars = 7
+    TargetCols = np.empty(nvars, dtype=np.int)
+    for i in range(len(headers)):
+        keys = ["logR", "density", "temperature", "velocity",
+                "ye", "pressure", "omega"]
+    TargetCols = [headers.index(key) for key in keys]
+
+    # GAMER-formatted array
+    GAMER_DATA_FORM = np.empty((0, nvars), dtype=np.float)
+    variables       = np.empty(nvars,      dtype=np.float)
+
+    for line in f:
+
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line.split()[icol])
+            # unit conversion
+            if idx == 0:
+                variables[idx] = R_sol * np.power(10, variables[idx]) # unit conversion: radius
+            if idx == 3:
+                variables[idx] = M2CM * variables[idx]                # unit conversion: velocity
+
+        GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM, variables])
+
+    GAMER_DATA_FORM = np.flipud(GAMER_DATA_FORM)
+
+    # add a safer line for the innermost radius
+    GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM[0], GAMER_DATA_FORM])
+    GAMER_DATA_FORM[0, 0] = 1.0
+
+    return f_info, GAMER_DATA_FORM
+
+
+if __name__ == "__main__":
+
+    # progenitor model format
+    WHS  = '1'
+    MESA = '2'
+
+    # load the command-line parameters
+    parser = argparse.ArgumentParser(description='Convert the progenitor model to GAMER-supported format\n')
+
+    parser.add_argument('-f', action='store', required=True, type=str, dest='format',
+                        help='original progenitor file format [1: WH07, Sukhbold2015, Heager2005 2: MESA] [%(default)s]')
```

Comment:
The order of `WH07, Sukhbold2015, Heager2005` is different than that in the file description

Reply:
I unified the order of description.